### ISSUE TITLE
Breadcrumbs initial state is empty when chained in multiple router outlets

### ISSUE THREAD

**Issue by Vankerkom** [udayvunnam/xng-breadcrumb#109]
### Description

The breadcrumbs do not initialize correctly when chained into multiple router outlets. The breadcrumbs show up after navigating to another page.

#### 🔬 Minimal Reproduction

Create an application with the following structure:

- Application Shell
  - Router Outlet
- Feature Shell (Lazy loaded)
  - Router Outlet
- Detailed Feature Shell
   - Breadcrumb Component
     - xng-breadcrumb

💻Your Environment

xng-breadcrumb@6.8.2
Angular Version: 12.1

## 🔥 Exception or Error

None.

**Comment by udayvunnam** [comment]
@Vankerkom can you please provide a reproducible example.

**Comment by TobiasXy** [comment]
For anyone who comes across this issue with a similar problem: I spend some time trying to create a reproducible example, because this happened on a project of mine too and it was super annoying.

Here it is: https://stackblitz.com/edit/angular-xng-breadcrumbs-ch2s2r?file=src%2Fapp%2Fapp.component.ts

What (apparently) was my problem:
I have an app with several layers of nested modules and a lots of lazy loaded routes. Take this structure for example (each entry is a module with corresponding component):

    app
        main <-- this extra level seems important to reproduce the problem
            feature1
            feature2
           
The <xng-breadcrumb> element is included in the "main" component, NOT in the "app" component.

When you open (or refresh) the page there is no breadcrumb shown. It only shows up once you click on "Feature2" and then it will stay even if you go back to Feature1. It's only missing on initial page load.

What solved it for me (and does as well in this example) is to require an instance of BreadcrumbService in AppComponent.
I guess this makes sure the service is instantiated early enough(?).

TL;DR:
Inject an instance of BreadcrumbService into the constructor of your AppComponent.

**Comment by udayvunnam** [comment]
Thank you @TobiasXy for the analysis and also for providing a fix from the application side.

If you know how to fix this in the library without having to inject in App component, please provide a fix as PR.

[contribution guide](https://github.com/udayvunnam/xng-breadcrumb/blob/main/.github/contributing.md)
[local development guide](https://udayvunnam.github.io/xng-breadcrumb/#/local-development-guide)

**Comment by udayvunnam** [comment]
not stale

**Comment by stephencarvalho** [comment]
I'm having the same issue. Using version 6.8.3 on Angular 10. Apart from the work around fix mentioned has there been any other fix?

**Comment by stephencarvalho** [comment]
@TobiasXy could you share a code example of the fix you mentioned. Because I injected the breadcrumbService into my app.component.ts but I'm still getting the same issue. 

When I initially load my application I do not see the breadcrumb but if I click on another page and comeback to the homepage then I can see the breadcrumbs

**Comment by stephencarvalho** [comment]
@udayvunnam Have there been any fixes for this issue?
